### PR TITLE
use puppet strings

### PR DESCRIPTION
**Pull request by Dan33l**
#### Pull Request (PR) description
This PR is a first step to use puppet strings.

A second second PR will be done after this one is merged. This will permit to switch from params.pp to common values in init.pp and OS specific values in hiera data of module. We will have auto generated default values in doc for common values.

By this way it split doc modification and code modification into dedicated PR.

### CODE REVIEW

File: manifests/hook.pp [hunk 1]
```diff
@@ -1,22 +1,16 @@
...
+# @param type Hook type.
+# @param hook_file Path to deploy hook script.
+# @param commands Bash commands to execute when the hook is run by certbot.
 #
 define letsencrypt::hook (
   Enum['pre', 'post', 'deploy'] $type,
   String[1]                     $hook_file,
   # hook.sh.epp will validate this
-  $commands,
+  Variant[String,Array[String]] $commands,
```
Comment (bastelfreak) [inline manifests/hook.pp]:
```suggestion
  Variant[String[1],Array[String[1]]] $commands,
```

**Comment by alexjfisher** [comment]
Tests are failing because of https://github.com/rspec/rspec-expectations/issues/1134

File: manifests/hook.pp (hunk 1 above)
Comment (Dan33l) [reply manifests/hook.pp]: updated

**Comment by Dan33l** [comment]
Closed because gems are cached.

**Pull request closed without merging**